tuxemon/save_upgrader.py

```diff
@@ -62,7 +65,7 @@
 }
 
 
-def upgrade_save(save_data: Dict[str, Any]) -> Dict[str, Any]:
+def upgrade_save(save_data: Dict[str, Any]) -> SaveData:
```

Comment:
Why isn't the input a `SaveData` too?

Reply:
save_upgrader a save file from an earlier format (i.e before refactoring) to the current format. The data that comes in might not match the schema of `SaveData`.

Reply:
Maybe then we should check that all fields are present and with appropriate types using a `TypeGuard`? We could use it if we add `typing_extensions` as a dependency, at least for type-checking.

Reply:
I'd rather not make this PR too complicated. Could you raise an issue explaining your idea?

Reply:
I've chosen to keep this PR simple, and tried to avoid functional changes. I've added some asserts, as you requested, but they were generally asserting that values that can't be none weren't none, in places where the game would immediately crash anyway.

Validating the save file is a bigger change, and there is some risk to doing it. I'd rather not break somebody's save file just because of some irrelevant type error.

If you're still interested then please raise a feature request.